common/lib/xmodule/xmodule/tests/test_util_adaptive_learning.py

```diff
@@ -369,17 +398,72 @@ def test_create_read_event(self):
             'type': 'EventRead',
             'payload': None,
         }
-        with patch.object(self.dummy_module, '_create_event') as patched__create_event:
+        with patch.object(DummyModule, '_create_event') as patched__create_event:
             patched__create_event.return_value = expected_event
             event = self.dummy_module.create_read_event(block_id, user_id)
             self.assertDictEqual(event, expected_event)
-            patched__create_event.assert_called_once_with(block_id, user_id, 'EventRead')
+            patched__create_event.assert_called_once_with(
+                adaptive_learning_configuration, block_id, user_id, 'EventRead'
+            )
+
+    @ddt.data(
+        ('correct', '100'), ('incorrect', '0')
+    )
+    @ddt.unpack
+    def test_create_result_event(self, success, payload):
+        """
+        Test that `create_result_event` method creates an event of type `EventResult` on external service,
+        and returns it.
+        """
+        course = self.dummy_module.parent_course
+        adaptive_learning_configuration = AdaptiveLearningConfiguration(**course.adaptive_learning_configuration)
+        block_id = 'knowledge-node-42'
+        user_id = 'student-42'
+        expected_event = {
+            'id': 42,
+            'knowledge_node_student_id': 42,
+            'type': 'EventResult',
+            'payload': payload,
+        }
+        with patch.object(DummyModule, '_make_anonymous_user_id') as patched__make_anonymous_user_id, \
+             patch.object(DummyModule, '_create_event') as patched__create_event:
+            patched__make_anonymous_user_id.return_value = user_id
```

Comment:
@itsjeyd I really don't like you mocking private methods/functions. This is a sign of dependency being hard-coded in a form of private method. Please either extract them into a proper form (i.e. new class that is injected into the object that uses it) or mock underlying implementation. I'd suggest the former.